src/TPPCore.ChatProviders/Providers/Twitch/TwitchIrcProvider.cs

```diff
@@ -52,12 +135,74 @@ private Task userNoticeEventHandler(IrcClient client, Message message)
+                foreach (string emote in emotes)
+                {
+                    int.TryParse(emote.Split(':')[0], out int emoteId); // : seperates the ID from the indexes
+                    string indexes = emote.Split(':')[1];
+                    if (emote.Contains(',')) // , seperates emote occurances
+                    {
+                        List<string> occurances = indexes.Split(',').ToList();
+                        foreach (string occurance in occurances)
+                        {
+                            int.TryParse(occurance.Split('-')[0], out int firstIndex); // - seperates the start index from the end index
+                            int.TryParse(occurance.Split('-')[1], out int secondIndex);
+                            string name = string.Empty;
+                            for (int i = firstIndex; i <= secondIndex; i++)
+                            {
+                                name += chatEvent.TextContent[i];
+                            }
+                            emoteOccurances.Add(new EmoteOccurance { Emotes = new Tuple<string, int, int>(name, firstIndex, secondIndex) });
+                            if (!keyValuePairs.ContainsKey(name))
+                            {
+                                keyValuePairs.Add(name, new Tuple<int, string>(emoteId, $"http://static-cdn.jtvnw.net/emoticons/v1/{emoteId}/3.0"));
+                            }
+                        }
+                    }
+                    else
+                    {
+                        int.TryParse(indexes.Split('-')[0], out int firstIndex);
+                        int.TryParse(indexes.Split('-')[1], out int secondIndex);
+                        string name = string.Empty;
+                        for (int i = firstIndex; i <= secondIndex; i++)
+                        {
+                            name += chatEvent.TextContent[i];
+                        }
+                        emoteOccurances.Add(new EmoteOccurance { Emotes = new Tuple<string, int, int>(name, firstIndex, secondIndex) });
+                        if (!keyValuePairs.ContainsKey(name))
+                        {
+                            keyValuePairs.Add(name, new Tuple<int, string>(emoteId, $"http://static-cdn.jtvnw.net/emoticons/v1/{emoteId}/3.0"));
+                        }
+                    }
+                }
+                chatEvent.Emote = new LoyaltyEvent.Emotes
+                {
+                    Data = keyValuePairs,
+                    Ranges = emoteOccurances
+                };
+                Debug.Assert(chatEvent.Emote.Ranges[0] != null);
+            }
+            else
+            {
+                chatEvent.Emote = new LoyaltyEvent.Emotes
+                {
+                    Data = new Dictionary<string, Tuple<int, string>> { },
+                    Ranges = new List<EmoteOccurance> { }
+                };
+                Debug.Assert(chatEvent.Emote != null);
+            }
```

Comment:
I think putting this into a function instead of copy and pasting might be better. 

(This is something you might do in a underhanded code contest to sneak code into things Kappa)